rfd/0017-supply-chain-hardening.md

```diff
@@ -0,0 +1,59 @@
+# Supply Chain Hardening
+
+To prevent supply chain attacks take control over all software distribution channels for Teleport and 
+only distribute signed software.
+
+## Why 
+
+Have a cryptographicaly-verifiable audit trail of all software changes and releases 
+and protect all software distribution channels.
+
+## How
+
+### Github
+
+Require [signed commits](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification) 
+for protected branches `master` and `branch/x.y`.
+
+Turn on commit signing for users and bots following steps 
+[here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification)
+
+**Github Mirror**
+
+Self host git repository for Teleport OSS, Enterprise, Cloud and webapps repositories.
+To allow outside collaboration, allow pull requests, forks, but process all commits using CI/CD pipeline 
+and do not use Github as a source of truth.
+
+**Teleport specific key repository**
+
+Establish Teleport-specific repository of supported public keys used by core team developers and have an 
+additional verification step done by Drone pipeline.
```

Comment:
What does drone do as verification?

Reply:
Presumably Drone would validate that each commit is signed by one of the authorised GPG keys which we maintain a list of.

Reply:
Drone should only allow merges with commits signed by trusted public keys.

Reply:
So all this key handling presumably comes into scope and should be specified should it not?